### PR TITLE
Initial JDBC Implementation

### PR DESCRIPTION
**Pull request by namirfa**

### CODE REVIEW

File: jdbcSource/README.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+In order to incorporate this Extension Source, you will need to set up your local machine with a JDBC Driver that can connect 
+to your SQL Database. Once you have done this, you will need to create the Source in the VANTIQ Modelo IDE. The documentation 
+has been split into two parts, [Setting Up Your Machine](#machine) and [Setting Up Your VANTIQ Modelo IDE](#vantiq).
+
+# Prerequisites <a name="pre" id="pre"></a>
+
+**IMPORTANT:** Read the [Testing](#testing) section before building this project.
+
+An understanding of the VANTIQ Extension Source SDK is assumed. Please read the Extension Source README.md for more
```
Comment (rblumer4) [inline jdbcSource/README.md]: would be good to make the README.md be a link to the readme (like the link in the next line)

File: jdbcSource/README.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+# Prerequisites <a name="pre" id="pre"></a>
+
+**IMPORTANT:** Read the [Testing](#testing) section before building this project.
+
+An understanding of the VANTIQ Extension Source SDK is assumed. Please read the Extension Source README.md for more 
+information.
+
+The user must [define the JDBC Source implementation](https://github.com/Vantiq/vantiq-extension-sources/blob/master/README.md#-defining-a-typeimplementation) in the VANTIQ Modelo IDE. For an example of the definition, 
+please see the *jbdcImpl.json* file located in the *src/test/resources* directory.
```
Comment (rblumer4) [inline jdbcSource/README.md]: this JSON file could also be made a link, I think

File: jdbcSource/README.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+*   **username**: Required. This is the username that will be used to connect to the SQL Database.
+*   **password**: Required. This is the password that will be used to connect to the SQL Database.
+*   **dbURL**: Required. This is the URL corresponding to the SQL Database that you will connect to.
+*   **pollRate**: Optional. If specified, you must specify the pollQuery as well. This option allows you to specify a polling 
+    rate indicating the frequency (in milliseconds) at which the pollQuery will be executed. The value must be a positive
+    number greater than 0, (*i.e.* 3000 --> executing every 3 seconds).
+*   **pollQuery**: Optional. If specified, you must specify the pollRate as well. This option indicates the SQL Query that
+    will be executed by the JDBC Source, (frequency assigned by the pollRate). The SQL Query must be a **SELECT** statement,
+    and the returned data will be sent as a Notification to the source. The data can be captured by creating a Rule in the
+    VANTIQ Modelo IDE, as per the following example:
```
Comment (rblumer4) [inline jdbcSource/README.md]: "as per" sounds odd here.  Maybe just "as in" ?

```suggestion
    VANTIQ Modelo IDE, as in the following example:
```

File: jdbcSource/README.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+}
+```
+
+## Publish Statements
+
+Another method to interact with the JDBC Source is to use VAIL to publish to the source. To do this, you will need to
+specify the SQL Query you wish to execute against your database as part of the Publish Parameters. The SQL Queries used here 
+can be **CREATE**, **INSERT**, **DELETE**, **DROP**, or other commands supported by your SQL database. The following is an 
+example of a Procedure created in VANTIQ Modelo publishing to a JDBC source.
+
```
Comment (rblumer4) [inline jdbcSource/README.md]: an example of an INSERT would be good here.
For example, can you pass an object to the INSERT with the values to be inserted?
Or do you have to create a query string with the full INSERT text??

File: jdbcSource/README.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    PUBLISH {"query":sqlQuery} to SOURCE JDBC1
+} catch (error) {
+    // Catching any errors and throwing the exception.
+    exception(error.code, error.message)
+}
+```
+
+## Error Messages
+
+Query errors originating from the source will always have the code be the FQCN with a small descriptor attached, and
```
Comment (rblumer4) [inline jdbcSource/README.md]: may need to spell out FQCN here
```suggestion
Query errors originating from the source will always have the code be the fully-qualified class name with a small descriptor attached, and
```

**Comment by namirfa** [comment]
Just cleaned up the README, let me know if there are any other comments. Also note that the link for the jdbcImpl.json file currently does not work, but will work once we squash this PR.

**Comment by namirfa** [comment]
*(Disregard previous comment about the link, changed links to be relative.)*

**Comment by fhcarter** [comment]
@namirfa Minor tweak -- in readme, the jdbcImpl.json link is spelled wrong -- jbdc... :-)

**Comment by namirfa** [comment]
Fixed!

**Pull request merged**